lib/events.js

```diff
@@ -332,12 +364,11 @@ EventEmitter.prototype.removeAllListeners =
 
       if (typeof listeners === 'function') {
         this.removeListener(type, listeners);
-      } else if (Array.isArray(listeners)) {
+      } else if (listeners && listeners.length) {
         // LIFO order
-        while (listeners.length)
+        while (listeners[0])
```

Comment:
Assignment to .length is slow, yes, but comparison shouldn't be.  I'm willing to believe that `listeners[0]` is faster (or maybe it's a wash) but I think the following two issues should be addressed:

1. The check in the `else if` clause is a little incongruous with the loop condition, it feels like it should be `} else if (listeners && listeners[0])`.

2. The loop looks like it can be turned into a do/while loop, skipping the first check.  Or you could keep it this way but reduce the `else if` clause to `} else if (listeners)`.